pkg/cmd/pr/checkout/checkout.go

```diff
@@ -268,3 +312,44 @@ func executeCmds(client *git.Client, credentialPattern git.CredentialPattern, cm
 	}
 	return nil
 }
+
+func selectPR(httpClient *http.Client, baseRepo ghrepo.Interface, prompter shared.Prompter, cs *iostreams.ColorScheme) (*api.PullRequest, error) {
+	listResult, err := list.ListPullRequests(httpClient, baseRepo, shared.FilterOptions{Entity: "pr", State: "open", Fields: []string{
+		"number",
+		"title",
+		"state",
+		"url",
+		"headRefName",
+		"headRepositoryOwner",
+		"isCrossRepository",
+		"isDraft",
+		"createdAt",
+	}}, 10)
+	if err != nil {
+		return nil, err
+	}
+	pr, err := promptForPR(prompter, cs, *listResult)
+	return pr, err
+}
+
+func promptForPR(prompter shared.Prompter, cs *iostreams.ColorScheme, jobs api.PullRequestAndTotalCount) (*api.PullRequest, error) {
+	candidates := []string{}
+	for _, pr := range jobs.PullRequests {
+		candidates = append(candidates, text.Truncate(120, fmt.Sprintf("%s %s (%s)",
+			shared.PRNumberWithColor(cs, pr),
+			text.RemoveExcessiveWhitespace(pr.Title),
+			cs.Gray(pr.HeadLabel()),
+		)))
+	}
+
+	selected, err := prompter.Select("Check out a specific PR?", "", candidates)
```

Comment:
This suggestion is to align this prompt text with some other "select a `<something>`" commands:

https://github.com/cli/cli/blob/9c550992ea4f207fa293cf0de00b633b47cff4c9/pkg/cmd/workflow/shared/shared.go#L201

https://github.com/cli/cli/blob/9c550992ea4f207fa293cf0de00b633b47cff4c9/pkg/cmd/gist/shared/shared.go#L241

https://github.com/cli/cli/blob/9c550992ea4f207fa293cf0de00b633b47cff4c9/pkg/cmd/pr/shared/templates.go#L202

I believe we also want to avoid abbreviating terms like `pull request` when they are part of a prompt or help message to make it [feel like GitHub](https://primer.style/native/cli/getting-started#make-it-feel-like-github).

```suggestion
	selected, err := prompter.Select("Select a pull request", "", candidates)
```

Reply:
Updated! thanks 🫡